src/analysis/vector/qgsgeometrysnapper.h

```diff
@@ -209,7 +209,7 @@ class QgsSnapIndex
         QgsPoint getSnapPoint( const QgsPoint &p ) const override;
         bool getIntersection( const QgsPoint &p1, const QgsPoint &p2, QgsPoint &inter ) const;
         bool getProjection( const QgsPoint &p, QgsPoint &pProj ) const;
-        bool withinDistance( const QgsPoint &p, const double distance );
+        bool withinSqrDistance( const QgsPoint &p, const double distance );
```

Comment:
Shouldn't this still be:

```suggestion
        bool withinDistance( const QgsPoint &p, const double distance );
```

since distance is the actual distance, not squared distance

Reply:
@nyalldawson , no, I'm passing on a squared tolerance value here (https://github.com/qgis/QGIS/pull/56426/commits/b1d22b9e0a6e8a2fe6c393817fb1319de44d9a21#diff-85f582115e6e43bb3410dea47308187c69ddece5c0621adf3f3aef770ef3cb60R268)

Reply:
Oh right, the diff was a little confusing there.

Could we go explicit with "bool withinSquaredDistance( const QgsPoint &p, const double squaredDistance );" ?

Reply:
@nyalldawson , yep, that's reasonable :wink: